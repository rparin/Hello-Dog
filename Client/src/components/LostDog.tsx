import Image from "next/image";
import Border from "@/components/Border";
import { lostDogPoster } from "@/constants/images";
import CountdownNavigateTo from "@/components/CountdownNavigateTo";

export default function LostDog() {
  return (
    <main>
      <Border className="relative flex h-full flex-col items-center justify-center gap-0 text-center">
        <Image
          fill={true}
          src={lostDogPoster.img}
          alt={lostDogPoster.alt}
          priority
        />
        <h2 className="absolute bottom-0 w-full bg-border pt-1 leading-none text-black/80">
          Returning to the home page in:
          <CountdownNavigateTo className="pl-1" href={"/"} countdown={5000} />
        </h2>
      </Border>
    </main>
  );
}
