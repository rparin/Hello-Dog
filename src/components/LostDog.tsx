import Link from "next/link";
import Image from "next/image";
import Border from "@/components/Border";
import { lostDogPoster } from "@/constants/images";
import CountdownNavigateTo from "@/components/CountdownNavigateTo";

export default function LostDog() {
  return (
    <main>
      <Border className="relative flex h-full flex-col items-center justify-center gap-0 text-center">
        <Image
          fill={true}
          src={lostDogPoster.img}
          alt={lostDogPoster.alt}
          priority
        />
        <h1 className="absolute bottom-0 w-full bg-border pt-1 leading-none text-black/80">
          <Link aria-label="Link to home page" className="underline" href={"/"}>
            Returning to the home page in:
          </Link>
          <CountdownNavigateTo
            className="pl-1 no-underline"
            href={"/"}
            countdown={5000}
          />
        </h1>
      </Border>
    </main>
  );
}
