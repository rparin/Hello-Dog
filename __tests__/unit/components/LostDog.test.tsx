import { render } from "@testing-library/react";
import { axe, toHaveNoViolations } from "jest-axe";
import LostDog from "@/components/LostDog";

expect.extend(toHaveNoViolations);

jest.mock("next/navigation", () => ({
  useRouter() {
    return {
      route: "/",
      pathname: "",
      query: "",
      asPath: "",
      push: jest.fn(),
      events: {
        on: jest.fn(),
        off: jest.fn(),
      },
      beforePopState: jest.fn(() => null),
      prefetch: jest.fn(() => null),
    };
  },
}));

test("it should have no accessibility violations", async () => {
  const { container } = render(<LostDog />);
  const results = await axe(container);
  expect(results).toHaveNoViolations();
});

test("it renders component unchanged", () => {
  const { container } = render(<LostDog />);
  expect(container).toMatchSnapshot();
});
