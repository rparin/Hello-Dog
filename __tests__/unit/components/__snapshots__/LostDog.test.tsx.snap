// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders component unchanged 1`] = `
<div>
  <main>
    <div
      class="h-dvh rounded-[30px] rounded-b-[50px] bg-border p-[10px] pb-[30px] md:rounded-[45px] md:rounded-b-[50px] md:p-[25px] md:pb-[50px]"
    >
      <div
        class="w-full overflow-auto rounded-[20px] rounded-b-[20px] bg-innerBg md:rounded-b-[10px] relative flex h-full flex-col items-center justify-center gap-0 text-center"
      >
        <img
          alt="Lost Dog poster"
          data-nimg="fill"
          decoding="async"
          fetchpriority="high"
          sizes="100vw"
          src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
        />
        <h2
          class="absolute bottom-0 w-full bg-border pt-1 leading-none text-black/80"
        >
          Returning to the home page in:
          <span
            class="pl-1"
          >
            5
          </span>
        </h2>
      </div>
      <section
        class="z-50 mx-auto flex w-full items-center justify-center gap-0.5 pt-1 text-black opacity-70 md:pt-3.5"
      >
        <p
          class="line-clamp-1 overflow-auto text-sm"
        >
          ©️ 2024. Designed and Developed by Ralph Parin.
        </p>
        <a
          aria-label="External link to github source code"
          href="https://github.com/rparin/Hello-Dog/tree/SpringBoot"
          rel="noopener noreferrer"
          target="_blank"
        >
          <svg
            class="lucide lucide-external-link mb-2 h-[.7rem] w-auto"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15 3h6v6"
            />
            <path
              d="M10 14 21 3"
            />
            <path
              d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"
            />
          </svg>
        </a>
      </section>
    </div>
  </main>
</div>
`;
